pkg/build/api/types.go

```diff
@@ -966,6 +987,9 @@ type BuildRequest struct {
 	// TriggeredBy describes which triggers started the most recent update to the
 	// buildconfig and contains information about those triggers.
 	TriggeredBy []BuildTriggerCause
+
+	// StrategyOptions contains additional strategy-specific options for the build
+	StrategyOptions *BuildStrategyOptions
```

Comment:
I've been implementing de-nesting this struct, but now anything that asks for a BuildStrategyOptions would need to ask for a SourceStrategyOptions, DockerStrategyOptions, etc. It's not an issue right now, but I think it will be a bit cumbersome in the future.

Maybe instead of having a containing struct (BuildStrategyOptions), we have an interface that each strategyoption implements?